datacapture/src/main/java/com/google/android/fhir/datacapture/fhirpath/ExpressionEvaluator.kt

```diff
@@ -81,6 +83,40 @@ object ExpressionEvaluator {
       }
   }
 
+  /**
+   * Returns a list of [Base] as the evaluated value for cqf expression extension. Expression runs
+   * over questionnaireResponse with fhirpath supplements values
+   * https://build.fhir.org/ig/HL7/sdc/expressions.html#fhirpath-supplements. %resource =
+   * [QuestionnaireResponse] %context = [QuestionnaireResponseItemComponent]
+   */
+  fun evaluateCqfExpression(
+    questionnaire: Questionnaire,
+    questionnaireResponse: QuestionnaireResponse,
+    questionnaireItem: QuestionnaireItemComponent,
+    questionnaireResponseItem: QuestionnaireResponseItemComponent,
+    expression: Expression,
+    questionnaireItemParentMap: Map<QuestionnaireItemComponent, QuestionnaireItemComponent>
+  ): List<Base> {
+    val appContext =
+      mutableMapOf<String, Base?>().apply {
+        extractDependentVariables(
+          expression,
+          questionnaire,
+          questionnaireResponse,
+          questionnaireItemParentMap,
+          questionnaireItem,
+          this
+        )
+      }
+    return fhirPathEngine.evaluate(
+      appContext,
+      questionnaireResponse,
+      null,
+      questionnaireResponseItem,
+      expression.expression
+    )
```

Comment:
optional : If you think existing code is more readable, then in that case you can please ignore this suggestion.

```suggestion
    val appContext = mutableMapOf<String, Base?>()
    return with(appContext) {
      extractDependentVariables(
        expression,
        questionnaire,
        questionnaireResponse,
        questionnaireItemParentMap,
        questionnaireItem,
        appContext
      )
      fhirPathEngine.evaluate(
        appContext,
        questionnaireResponse,
        null,
        questionnaireResponseItem,
        expression.expression
      )
    }
```